src/main/java/it/pagopa/transactions/commands/handlers/v2/TransactionRequestUserReceiptHandler.java

```diff
@@ -104,62 +137,111 @@ public Mono<BaseTransactionEvent<?>> handle(TransactionAddUserReceiptCommand com
-                            new it.pagopa.ecommerce.commons.documents.v2.TransactionUserReceiptData(
-                                    requestOutcomeToReceiptOutcome(
-                                            command.getData().addUserReceiptRequest().getOutcome()
-                                    ),
-                                    language,
-                                    addUserReceiptRequestDto.getPaymentDate().toZonedDateTime().toString()
-                            )
-                    );
-
-                    return userReceiptAddedEventRepository.save(event)
-                            .flatMap(
-                                    userReceiptEvent -> tracingUtils.traceMono(
-                                            this.getClass().getSimpleName(),
-                                            tracingInfo -> transactionNotificationRequestedQueueAsyncClient
-                                                    .sendMessageWithResponse(
-                                                            new QueueEvent<>(userReceiptEvent, tracingInfo),
-                                                            Duration.ZERO,
-                                                            Duration.ofSeconds(transientQueuesTTLSeconds)
-                                                    )
-                                    ).doOnError(
-                                            exception -> log.error(
-                                                    "Error to generate event {} for transactionId {} - error {}",
-                                                    event.getEventCode(),
-                                                    event.getTransactionId(),
-                                                    exception.getMessage()
-                                            )
+                    return Tuples.of(
+                            new it.pagopa.ecommerce.commons.documents.v2.TransactionUserReceiptRequestedEvent(
+                                    command.getData().transactionId().value(),
+                                    new it.pagopa.ecommerce.commons.documents.v2.TransactionUserReceiptData(
+                                            requestOutcomeToReceiptOutcome(
+                                                    command.getData().addUserReceiptRequest().getOutcome()
+                                            ),
+                                            language,
+                                            addUserReceiptRequestDto.getPaymentDate().toZonedDateTime().toString()
                                     )
-                                            .doOnNext(
-                                                    queueResponse -> log.info(
+                            ),
+                            tx
+                    );
+                }).flatMap(
+                        event_tx -> userReceiptAddedEventRepository.save(event_tx.getT1())
```

Comment:
If you want to avoid doing `event_tx.getT1()` and so on you could refactor this into

```suggestion
                        TupleUtils.function((event, tx) -> userReceiptAddedEventRepository.save(event)
```